src/main/java/org/radarcns/consumer/realtime/RealtimeInferenceConsumer.java

```diff
@@ -0,0 +1,185 @@
+package org.radarcns.consumer.realtime;
+
+import static io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.CLIENT_ID_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_ID_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
+
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.errors.WakeupException;
+import org.radarbase.util.RollingTimeAverage;
+import org.radarcns.config.ConfigRadar;
+import org.radarcns.config.realtime.RealtimeConsumerConfig;
+import org.radarcns.consumer.realtime.action.Action;
+import org.radarcns.consumer.realtime.action.ActionFactory;
+import org.radarcns.consumer.realtime.condition.Condition;
+import org.radarcns.consumer.realtime.condition.ConditionFactory;
+import org.radarcns.monitor.KafkaMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The main Kafka Consumer class that runs a single consumer on any topic. The consumer evaluates
+ * each incoming record based on the {@link Condition}s provided in the config. If and only If all
+ * the conditions evaluate to true, only then all the configured {@link Action}s are fired.
+ *
+ * <p>To be used with the model-invocation-endpoint and KSQL API_INFERENCE function to evaluate and
+ * take action on incoming results from realtime inference on data.
+ */
+public class RealtimeInferenceConsumer implements KafkaMonitor {
```

Comment:
If I understand it correctly, all of these Consumers are realtime. We can remove that word from the class name I think.
Why not reuse/extend from existing AbstractKafkaMonitor?

Reply:
I think the adjective realtime was added for inference and not for consumers. So these are consumers for realtime inference as compared to batch or historic inference. But I agree if it seems confusing best to remove it. Just to allow more flexibility (like for now a different constructor spec) but can definitely extend from AbstractKafkaMonitor if we need functionality already present there.